cups/file.c

```diff
@@ -335,6 +335,131 @@ _cupsFileCheckFilter(
 #endif /* !_WIN32 */
 
 
+/*
+ * 'cupsFileGetConfAndComments()' - Get line and comments from a configuration file.
+ */
+
+char *					/* O  - Line read or @code NULL@ on end of file or error */
+_cupsFileGetConfAndComments(cups_file_t *fp,	/* I  - CUPS file */
+                char        *buf,	/* O  - String buffer */
+		size_t      buflen,	/* I  - Size of string buffer */
+                char        **value,	/* O  - Pointer to value */
+		int         *linenum)	/* IO - Current line number */
+{
+  char	*ptr;				/* Pointer into line */
+
+
+ /*
+  * Range check input...
+  */
+
+  DEBUG_printf(("2cupsFileGetConfAndComments(fp=%p, buf=%p, buflen=" CUPS_LLFMT
+                ", value=%p, linenum=%p)", (void *)fp, (void *)buf, CUPS_LLCAST buflen, (void *)value, (void *)linenum));
+
+  if (!fp || (fp->mode != 'r' && fp->mode != 's') ||
+      !buf || buflen < 2 || !value)
+  {
+    if (value)
+      *value = NULL;
+
+    return (NULL);
+  }
+
+ /*
+  * Read the next line...
+  */
+
+  *value = NULL;
+
+  while (cupsFileGets(fp, buf, buflen))
+  {
+    (*linenum) ++;
+
+    /*
+     * Strip the comment from inline comment...
+     */
+     if ((ptr = strchr(buf, '#')) != NULL)
+     {
+       int index = (int) (ptr - buf);
+       while (index > 0 && _cups_isspace(buf[index]))
+         buf[index--] = '\0';
```

Comment:
1. setting the NULL terminator in every iteration is redundant - set it only once you know there is a non-whitespace character before #.
2. have you tested the code for a line "       # comment"? What happens for such a line with the current code?
3. what will happen if escaped # appears? - \# - this is not a comment.
4. do you think this can be solved with pointer arithmetic instead of introducing a new variable ```index```?

Help: look at cupsFileGetConf() solves it, think about what is needed in our case, adjust it accordingly and use it here.